substrafl_examples/get_started/plot_substrafl_sklearn_fedavg.py

```diff
@@ -0,0 +1,523 @@
+# :ref:`substrafl_doc/api/algorithms:Base Class`
+#
+# To define a custom algorithm, we will need to inherit from the base class Algo, and to define 2 properties and 4
+# functions:
+#
+# - **strategies** (property): the list of strategy name our algorithm is compatible with.
+# - **model** (property): a property that return the algo model.
+# - **train** (function): a function with the @remote_data decorator. This function describe the training process to
+#   apply to train our model in a federated way.
+# - **predict** (function): a function with the @remote_data decorator. This function describe how to compute the
+#   predictions from the algo model.
+# - **save** (function): specify how to save the important states of our algo.
+# - **load** (function): specify how to load the important states of our algo from a previously saved filed
+#   by the `save` function describe above.
+
+# Number of model update between each FL strategy aggregation.
+from substrafl import algorithms
+from substrafl import remote
+from substrafl import schemas as fl_schemas
+
+import joblib
+from typing import Optional
+import shutil
+
+class SklearnFedAvgAlgo(algorithms.Algo):
+    def __init__(self, model, seed=None):
+        super().__init__(model=model, seed=seed)
+
+        self._model = model
+
+        if seed is not None:
+            np.random.seed(seed)
+
+    @property
+    def strategies(self):
+        """List of compatible strategies"""
+        return [fl_schemas.StrategyName.FEDERATED_AVERAGING]
+
+    @property
+    def model(self):
+        return self._model
+
+    @remote.remote_data
+    def train(
+        self,
+        datasamples,
+        shared_state: Optional[fl_schemas.FedAvgAveragedState] = None,
+    ) -> fl_schemas.FedAvgSharedState:
+
+        if shared_state is None:
+            # If shared state is None, we are at the init state of the algorithm.
+            # We need all different instance of the algorithm to have the same initialization.
+            self._model.coef_ = np.ones((3, 4))
+            self._model.intercept_ = np.zeros(3)
+        else:
+            # If we have a shared state, we update the model parameters with the average parameters
+            # updates.
+            self._model.coef_ += np.reshape(shared_state.avg_parameters_update[:-1], (3, 4))
+            self._model.intercept_ += shared_state.avg_parameters_update[-1]
+
+        # To be able to compute the delta between the parameters before and after training, we need to save
+        # them in a temporary variable.
+        old_coef = self._model.coef_
+        old_bias = self._model.intercept_
+
+        # Model training.
+        self._model.fit(datasamples["data"], datasamples["targets"])
+
+        # We compute de delta.
+        delta_coef = self._model.coef_ - old_coef
+        delta_bias = self._model.intercept_ - old_bias
+
+        # We reset the model parameters to their state before training in order to remove the local updates from it.
```

Comment:
We have to scroll from left to right to read it.
Either make the lines smaller or we increase the line-length a bit. What do you think?

Reply:
I still have to scroll from left to right. What do you think we should do?

Reply:
My bad, I didn't verify properly after the modif. It should be ok now.